From: Gavin Schenk <g.schenk@eckelmann.de>
Date: Thu, 23 Nov 2017 12:13:48 +0100
Subject: [PATCH] fbv: Fix build error due to direct access to jmpbuf

The build fails with "dereferencing pointer to incomplete type" when
accessing the jmpbuf ptr directly.

Using the macro png_jmpbuf(...) fixes this compiler error.
---
 png.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/png.c b/png.c
index 8a0c62073dfb..3f5afddbca5c 100644
--- a/png.c
+++ b/png.c
@@ -69,7 +69,7 @@ int fh_png_load(char *name,unsigned char *buffer, unsigned char ** alpha,int x,i
         fclose(fh); return(FH_ERROR_FORMAT);
     }
     rp=0;
-    if (setjmp(png_ptr->jmpbuf))
+    if (setjmp(png_jmpbuf(png_ptr)))
     {
 	png_destroy_read_struct(&png_ptr, &info_ptr, (png_infopp)NULL);
         if(rp) free(rp);
@@ -161,7 +161,7 @@ int fh_png_getsize(char *name,int *x,int *y)
         fclose(fh); return(FH_ERROR_FORMAT);
     }
     rp=0;
-    if (setjmp(png_ptr->jmpbuf))
+    if (setjmp(png_jmpbuf(png_ptr)))
     {
 	png_destroy_read_struct(&png_ptr, &info_ptr, (png_infopp)NULL);
         if(rp) free(rp);
